input_files/GIT_team_testing/IRA900_selenate_fixedbed.i

```diff
@@ -0,0 +1,1361 @@
+        variable = q_Cl5
+        this_variable = q_Cl5
+        forward_rate = 1E4      ######to be determined#######
+        reverse_rate = 0
+        scale = -2.0
+        reactants = 'Cp_SeO45 q_Cl5'
+        reactant_stoich = '1 2'
+        products = 'q_SeO45'
+        product_stoich = '1'
+    [../]
+
+
+    [./q_Cl6_dot]
+        type = TimeDerivative
+        variable = q_Cl6
+    [../]
+    [./q_Cl6_rxn]
+#   SeO46 + 2R-SeO46 <-- --> R-SeO46 + 2Cl6
+#   Cp_SeO46  q_Cl6        q_SeO46 + Cp_Cl6
+        type = ConstReaction
+        variable = q_Cl6
+        this_variable = q_Cl6
+        forward_rate = 1E4      ######to be determined#######
+        reverse_rate = 0
+        scale = -2.0
+        reactants = 'Cp_SeO46 q_Cl6'
+        reactant_stoich = '1 2'
+        products = 'q_SeO46'
+        product_stoich = '1'
+    [../]
+
+
+    [./q_Cl7_dot]
+        type = TimeDerivative
+        variable = q_Cl7
+    [../]
+    [./q_Cl7_rxn]
+#   SeO47 + 2R-Cl7 <-- --> R-SeO47 + 2Cl7
+#   Cp_SeO47  q_Cl7        q_SeO47 + Cp_Cl7
+        type = ConstReaction
+        variable = q_Cl7
+        this_variable = q_Cl7
+        forward_rate = 1E4      ######to be determined#######
+        reverse_rate = 0
+        scale = -2.0
+        reactants = 'Cp_SeO47 q_Cl7'
+        reactant_stoich = '1 2'
+        products = 'q_SeO47'
+        product_stoich = '1'
+    [../]
+
+
+    [./q_Cl8_dot]
+        type = TimeDerivative
+        variable = q_Cl8
+    [../]
+    [./q_Cl8_rxn]
+#   SeO48 + 2R-Cl8 <-- --> R-SeO48 + 2Cl8
+#   Cp_SeO48  q_Cl8        q_SeO48 + Cp_Cl8
+        type = ConstReaction
+        variable = q_Cl8
+        this_variable = q_Cl8
+        forward_rate = 1E4      ######to be determined#######
+        reverse_rate = 0
+        scale = -2.0
+        reactants = 'Cp_SeO48 q_Cl8'
+        reactant_stoich = '1 2'
+        products = 'q_SeO48'
+        product_stoich = '1'
+    [../]
+
+
+    [./q_Cl9_dot]
+        type = TimeDerivative
+        variable = q_Cl9
+    [../]
+    [./q_Cl9_rxn]
+#   SeO49 + 2R-Cl9 <-- --> R-SeO49 + 2Cl9
+#   Cp_SeO49  q_Cl9        q_SeO49 + Cp_Cl9
+        type = ConstReaction
+        variable = q_Cl9
+        this_variable = q_Cl9
+        forward_rate = 1E4      ######to be determined#######
+        reverse_rate = 0
+        scale = -2.0
+        reactants = 'Cp_SeO49 q_Cl9'
+        reactant_stoich = '1 2'
+        products = 'q_SeO49'
+        product_stoich = '1'
+    [../]
+
+
+ []
+
+## Forgot to add DG kernels
+[DGKernels]
+		[./SeO4Adv_dg]
+			 type = DGPoreConcAdvection
+			 variable = Cb_SeO4
+			 porosity = eps_b
+			 ux = vel_x
+			 uy = vel_y
+			 uz = vel_z
+		[../]
+		[./SeO4Diff_dg]
+				type = DGVarPoreDiffusion
+				variable = Cb_SeO4
+				porosity = eps_b
+				Dx = Db_SeO4
+				Dy = Db_SeO4 #axial dispersion cm^2/min (mostly for stabilization - need to check to see if valid. Otherwise, use 'order = CONSTANT' for stabilization)
+				Dz = Db_SeO4
+		[../]
+[]
```

Comment:
Ziheng, these kernels were left out in your file which is what caused the odd simulation results. When you use 'MONOMIAL' as the 'family' for your variables, then you must include 'DGKernels' for corresponding advection and diffusion components in the 'Kernels' section. Failure to do so results in no transfer of mass between elements. Under that circumstance, you have flux in from the boundary but not leaving the first element which results in a massive 'ripple' in the first element of very large positive and very large negative values. 

Additionally, for any 'DGKernels', you must also provide a variable for the velocity (ux, uy, and uz) and diffusion (Dx, Dy, and Dz) terms. This is because the kernel expects to find variables in the neighboring cells and will query for them. If you provide a constant value (i.e., 0.099 or similar) in these locations, it will result in a Segfault error.